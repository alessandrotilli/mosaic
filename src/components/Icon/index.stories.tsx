import React from "react";
import MUIStyleIcon from "@mui/icons-material/Style";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Icons } from "../../types/Icon";
import getDocsPage from "../../utils/stories";

import Icon, { DATA_CY_DEFAULT } from ".";

export default {
  title: "Display/Icon",
  component: Icon,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Icon Component",
          url: "https://v4.mui.com/components/icons/",
        },
        component: "Icon",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
        },
      }),
    },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} dataCy={DATA_CY_DEFAULT} />;

export const Primary = Template.bind({});
Primary.args = {
  name: Icons.account,
};

export const IconCustom = Template.bind({});
IconCustom.args = {
  children: <MUIStyleIcon />,
};

export const Loading = Template.bind({});
Loading.args = {
  ...Primary.args,
  loading: true,
};

export const Rotate = Template.bind({});
Rotate.args = {
  ...Primary.args,
  rotate: true,
};

export const SizeSmall = Template.bind({});
SizeSmall.args = {
  ...Primary.args,
  size: "small",
};

export const SizeLarge = Template.bind({});
SizeLarge.args = {
  ...Primary.args,
  size: "large",
};

export const Styled = Template.bind({});
Styled.args = {
  ...Primary.args,
  style: {
    backgroundColor: "red",
    borderRadius: "4px",
    color: "white",
    padding: "4px",
  },
};

// TODO: add this story using union type
// const allIcons = Object.values(Icons);
// const getIcon = (icon: string) => allIcons.find((i) => i === icon);
// const allIconsJsx = allIcons.map((icon, index) => (
//   <div key={index} className="icon-wrapper">
//     <Icon dataCy={`icon-${icon}`} name={getIcon(icon) || Icons.add} />
//     <span>{icon}</span>
//   </div>
// ));

// export const AllIcons = () => <StoriesWrapper>{allIconsJsx}</StoriesWrapper>;
