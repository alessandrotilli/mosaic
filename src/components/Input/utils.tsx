import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
