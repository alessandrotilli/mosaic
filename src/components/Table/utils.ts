export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
