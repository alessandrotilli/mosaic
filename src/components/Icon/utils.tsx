import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
