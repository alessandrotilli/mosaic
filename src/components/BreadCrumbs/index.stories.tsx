import React from "react";
import AccessibilityIcon from "@mui/icons-material/Accessibility";
import { Meta, StoryObj } from "@storybook/react";
import { configure, expect, fireEvent, fn, within } from "@storybook/test";

import { Icons } from "../../types/Icon";
import { getComposedDataCy } from "../../utils";
import { logInfo } from "../../utils/logger";
import getDocsPage from "../../utils/stories";
import LocaleDecorator from "../../utils/stories/decorators/Locale";

import { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, LocalizedBreadCrumbs, SUBPARTS_MAP } from ".";

configure({ testIdAttribute: "data-cy" });

const COMPONENT_NAME = "BreadCrumbs";
LocalizedBreadCrumbs.displayName = COMPONENT_NAME;

const meta = {
  title: "Navigation/BreadCrumbs",
  component: LocalizedBreadCrumbs,
  decorators: [LocaleDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI BreadCrumbs Component",
          url: "https://mui.com/material-ui/react-breadcrumbs/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
    },
  },
} satisfies Meta<typeof LocalizedBreadCrumbs>;

export default meta;
type Story = StoryObj<typeof meta>;

const linksMock = [
  { label: "page1", href: "//google.com/page1" },
  { label: "page2", href: "//google.com/page2" },
  { label: "page3", href: "//google.com/page3" },
];

const onClickMock = fn(() => logInfo(COMPONENT_NAME, "onClick handler"));

export const Primary: Story = {
  args: {
    dataCy: DATA_CY_DEFAULT,
    links: linksMock,
    onClick: onClickMock,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step("First Link", async () => {
      const firstIndex = 0;
      const firstLinkDataCy = getComposedDataCy(DATA_CY_DEFAULT, SUBPARTS_MAP.links, firstIndex);
      const firstLink = canvas.getAllByTestId(firstLinkDataCy).at(0);
      if (!firstLink) {
        return;
      }

      fireEvent.click(firstLink);
      await expect(onClickMock).toHaveBeenCalledTimes(onClickMock.mock.calls.length);
      await expect(onClickMock).toHaveBeenCalledWith(linksMock[firstIndex].href);
    });

    await step("Last Link", async () => {
      const lastIndex = linksMock.length - 1;
      const lastLinkDataCy = getComposedDataCy(DATA_CY_DEFAULT, SUBPARTS_MAP.links, lastIndex);
      const lastLink = canvas.getAllByTestId(lastLinkDataCy).at(0);
      if (!lastLink) {
        return;
      }

      fireEvent.click(lastLink);
      await expect(onClickMock).toHaveBeenCalledTimes(onClickMock.mock.calls.length);
    });
  },
};

export const CustomSeparator: Story = {
  args: {
    ...Primary.args,
    separator: ">",
  },
};

export const SizeMedium: Story = {
  args: {
    ...Primary.args,
    size: "medium",
  },
};

export const SizeLarge: Story = {
  args: {
    ...Primary.args,
    size: "large",
  },
};

export const WithIcons: Story = {
  args: {
    ...Primary.args,
    links: linksMock.map((link, index) => ({ ...link, icon: index % 2 === 0 ? Icons.clock : <AccessibilityIcon /> })),
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    links: linksMock.map((link) => ({ ...link, label: `locale.${link.label}` })),
  },
};
