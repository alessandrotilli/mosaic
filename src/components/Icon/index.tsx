import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
