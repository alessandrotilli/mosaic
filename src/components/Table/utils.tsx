import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
