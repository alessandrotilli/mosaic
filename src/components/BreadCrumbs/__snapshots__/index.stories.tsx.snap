// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/BreadCrumbs CustomSeparator smoke-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            page1
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      &gt;
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            page2
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      &gt;
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          page3
        </h3>
      </div>
    </li>
  </ol>
</nav>
`;

exports[`Navigation/BreadCrumbs Localized smoke-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            Pagina 1
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            Pagina 2
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          Pagina 3
        </h3>
      </div>
    </li>
  </ol>
</nav>
`;

exports[`Navigation/BreadCrumbs Primary play-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            page1
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            page2
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          page3
        </h3>
      </div>
    </li>
  </ol>
</nav>
`;

exports[`Navigation/BreadCrumbs SizeLarge smoke-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <h1 class="MuiTypography-root MuiTypography-h5 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            page1
          </h1>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <h1 class="MuiTypography-root MuiTypography-h5 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            page2
          </h1>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <h1 class="MuiTypography-root MuiTypography-h5 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          page3
        </h1>
      </div>
    </li>
  </ol>
</nav>
`;

exports[`Navigation/BreadCrumbs SizeMedium smoke-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <h2 class="MuiTypography-root MuiTypography-h6 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            page1
          </h2>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <h2 class="MuiTypography-root MuiTypography-h6 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            page2
          </h2>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <h2 class="MuiTypography-root MuiTypography-h6 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          page3
        </h2>
      </div>
    </li>
  </ol>
</nav>
`;

exports[`Navigation/BreadCrumbs WithIcons smoke-test 1`] = `
<nav class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-mui-classname"
     data-cy="breadcrumbs"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol css-mui-classname">
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-0"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-0"
             style="align-items: center; display: flex;"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="ScheduleIcon"
               data-cy="breadcrumbs-link-0-icon"
               style="margin-right: 8px;"
          >
            <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8">
            </path>
            <path d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z">
            </path>
          </svg>
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-0-label"
          >
            page1
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-mui-classname"
         data-cy="breadcrumbs-link-1"
         style="align-items: center; display: flex;"
      >
        <div data-cy="breadcrumbs-link-1"
             style="align-items: center; display: flex;"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="AccessibilityIcon"
               data-cy="breadcrumbs-link-1-icon"
               style="margin-right: 8px; height: 20px; width: 20px;"
          >
            <path d="M12 2c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m9 7h-6v13h-2v-6h-2v6H9V9H3V7h18z">
            </path>
          </svg>
          <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
              data-cy="breadcrumbs-link-1-label"
          >
            page2
          </h3>
        </div>
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator css-mui-classname"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <div data-cy="breadcrumbs-link-2"
           style="align-items: center; display: flex;"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="ScheduleIcon"
             data-cy="breadcrumbs-link-2-icon"
             style="margin-right: 8px;"
        >
          <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8">
          </path>
          <path d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z">
          </path>
        </svg>
        <h3 class="MuiTypography-root MuiTypography-subtitle1 css-mui-classname"
            data-cy="breadcrumbs-link-2-label"
        >
          page3
        </h3>
      </div>
    </li>
  </ol>
</nav>
`;
